cub/device/dispatch/dispatch_batch_memcpy.cuh

```diff
@@ -0,0 +1,631 @@
+
+  // Block-level specialization
+  using AgentBatchMemcpyT = AgentBatchMemcpy<AgentBatchMemcpyPolicyT,
+                                             InputBufferIt,
+                                             OutputBufferIt,
+                                             BufferSizeIteratorT,
+                                             BufferOffsetT,
+                                             BlevBufferSrcsOutItT,
+                                             BlevBufferDstsOutItT,
+                                             BlevBufferSizesOutItT,
+                                             BlevBufferTileOffsetsOutItT,
+                                             BlockOffsetT,
+                                             BLevBufferOffsetTileState,
+                                             BLevBlockOffsetTileState>;
+
+  // Shared memory for AgentBatchMemcpy
+  __shared__ typename AgentBatchMemcpyT::TempStorage temp_storage;
+
+  // Process this block's tile of input&output buffer pairs
+  AgentBatchMemcpyT(temp_storage,
+                    input_buffer_it,
+                    output_buffer_it,
+                    buffer_sizes,
+                    num_buffers,
+                    blev_buffer_srcs,
+                    blev_buffer_dsts,
+                    blev_buffer_sizes,
+                    blev_buffer_tile_offsets,
+                    blev_buffer_scan_state,
+                    blev_block_scan_state)
+    .ConsumeTile(blockIdx.x);
+}
+
+template <typename BufferOffsetT, typename BlockOffsetT>
+struct DeviceBatchMemcpyPolicy
+{
+  static constexpr uint32_t BLOCK_THREADS         = 128U;
+  static constexpr uint32_t BUFFERS_PER_THREAD    = 4U;
+  static constexpr uint32_t TLEV_BYTES_PER_THREAD = 8U;
+
+  static constexpr uint32_t LARGE_BUFFER_BLOCK_THREADS    = 256U;
+  static constexpr uint32_t LARGE_BUFFER_BYTES_PER_THREAD = 32U;
+  // TODO make true default for pre-Volta, false for Volta and after
+  static constexpr bool PREFER_POW2_BITS = false;
+
+  /// SM35
+  struct Policy350 : ChainedPolicy<350, Policy350, Policy350>
+  {
+    using AgentSmallBufferPolicyT = AgentBatchMemcpyPolicy<BLOCK_THREADS,
+                                                           BUFFERS_PER_THREAD,
+                                                           TLEV_BYTES_PER_THREAD,
+                                                           PREFER_POW2_BITS,
+                                                           LARGE_BUFFER_BLOCK_THREADS * LARGE_BUFFER_BYTES_PER_THREAD>;
+
+    using AgentLargeBufferPolicyT =
+      AgentBatchMemcpyLargeBuffersPolicy<LARGE_BUFFER_BLOCK_THREADS, LARGE_BUFFER_BYTES_PER_THREAD>;
+  };
+
+  using MaxPolicy = Policy350;
+};
+
+/**
+ * @tparam InputBufferIt <b>[inferred]</b> Random-access input iterator type providing the pointers to the source
+ * memory buffers
+ * @tparam OutputBufferIt <b>[inferred]</b> Random-access input iterator type providing the pointers to the
```

Comment:
Similar to https://github.com/NVIDIA/cub/pull/359#discussion_r871362334, I'm inclined to keep calling this `random-access input iterator type providing the pointers` as we only read from this iterator and don't write to the iterator itself. But let me know if you think otherwise.